### PR TITLE
feat(server): add liveReload option to enable/disable live reloading

### PR DESCRIPTION
**Pull request by EslamHiko**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
yes

### Motivation / Use-Case
fix this pr : https://github.com/webpack/webpack-dev-server/pull/1812

### Breaking Changes
no

### Additional Info
I hope everything works fine this time.

### CODE REVIEW

File: test/ContentBase.test.js [hunk 1]
```diff
@@ -18,6 +18,109 @@ const contentBaseOther = path.join(
...
+
+        done();
+      });
+
+      // change file content
+      setTimeout(() => {
+        fs.writeFileSync(nestedFile, 'Heyo', 'utf8');
+      }, 1000);
+    });
+  });
```
Comment (evilebottnawi) [inline test/ContentBase.test.js]: Let's move this test to own test file `LiveReload.test.js`

File: lib/Server.js [hunk 2]
```diff
@@ -95,6 +95,7 @@ class Server {
     this.hot = this.options.hot || this.options.hotOnly;
     this.headers = this.options.headers;
     this.progress = this.options.progress;
+    this.liveReload = this.options.liveReload;
```
Comment (evilebottnawi) [inline lib/Server.js]: Don't add here new options, you can see https://github.com/webpack/webpack-dev-server/blob/0523cfd0309f7add440169b523176ed7baa64ec3/lib/Server.js#L94 use `this.options.liveReload` everywhere

File: lib/Server.js [hunk 3]
```diff
@@ -719,6 +720,10 @@ class Server {
         this.sockWrite([connection], 'hot');
       }
 
+      if (this.liveReload !== false) {
```
Comment (evilebottnawi) [inline lib/Server.js]: As written above please use `this.options.liveReload`

File: lib/Server.js [hunk 4]
```diff
@@ -999,11 +1004,12 @@ class Server {
     };
 
     const watcher = chokidar.watch(watchPath, watchOptions);
-
-    watcher.on('change', () => {
-      this.sockWrite(this.sockets, 'content-changed');
-    });
-
+    // disabling refreshing on changing the content
+    if (this.liveReload !== false) {
```
Comment (evilebottnawi) [inline lib/Server.js]: As written above please use `this.options.liveReload`

**Comment by EslamHiko** [comment]
@evilebottnawi is this problem has something to do with the snapshots?

**Comment by evilebottnawi** [comment]
@EslamHiko should not

**Comment by hiroppy** [comment]
Thanks

**Comment by kaiyoma** [comment]
Thanks so much for doing this! I just tested out the new version and everything works as I expect.

**Pull request merged**